app/views/result_filters/location/new.html.erb

```diff
@@ -14,31 +14,76 @@
         </legend>
 
         <div class="govuk-form-group">
-          <% if flash[:error] %>
-            <span id="location-error" class="govuk-error-message">
-              <span class="govuk-visually-hidden">Error:</span> Please choose an option
-            </span>
-          <% end %>
-          <div class="govuk-radios">
+          <div class="govuk-radios govuk-radios--conditional" data-module="govuk-radios">
             <div class="govuk-radios__item">
               <%=
                 form.radio_button(
                   :l,
-                  "2",
+                  "1",
                   class: "govuk-radios__input",
-                  data: { qa: "across_england" },
-                  checked: params[:l] == "2"
+                  data: { qa: "by_postcode_town_or_city" },
+                  checked: params[:l] == "1",
+                  aria: { 'controls': "location-conditional" }
                 )
               %>
-              <%= form.label :l_2, "Across England", class: "govuk-label govuk-radios__label" %>
+              <%= form.label :l_1, "By postcode, town or city", class: "govuk-label govuk-radios__label" %>
+            </div>
+            <div class="govuk-radios__conditional <%= "govuk-radios__conditional--hidden" unless flash[:location_error] %>" id="location-conditional">
+              <div class="govuk-form-group">
+                <%= form.label :lq, "Postcode, town or city", class: "govuk-label" %>
+                <div class="govuk-error-message" id="location-error" data-qa="location-error"><%= flash[:location_error] %></div>
+                <%=
+                  form.text_field(
+                    :lq,
+                    value: "#{flash[:query]}",
+                    class: "govuk-input",
+                    data: { qa: "location-query" },
+                    )
+                %>
+              </div>
+              <div class="govuk-form-group">
+                <%= form.label :rad, "Within", class: "govuk-label" %>
+                <%= form.select :rad,
+                                options_for_select(
+                                  [
+                                    ["5 miles", "5"],
+                                    ["10 miles", "10"],
+                                    ["20 miles", "20"],
+                                    ["50 miles", "50"],
+                                    ["100 miles", "100"],
+                                  ],
+                                  ["20"]
```

Comment:
will 20 be overriden by the user when they revisit to up the value?

ie

```
user tried 100 miles, comes back and down it to 50 (what value they see before change 20 or 100?)
```

Reply:
Fixed 👍